zipnumcluster-ccpyspark.py

```diff
@@ -0,0 +1,263 @@
+        # Handle final chunk
+        if current_chunk:
+            chunk_data = ''.join(current_chunk).encode('utf-8')
+            z = zlib.compressobj(6, zlib.DEFLATED, zlib.MAX_WBITS + 16)
+            compressed = z.compress(chunk_data) + z.flush()
+            chunk_length = len(compressed)
+            f.write(compressed)
+            
+            index_entries.append((
+                str(chunk_min_surt),  # min surt
+                str(chunk_max_surt),  # max surt
+                str(output_filename),
+                int(partition_id),
+                int(current_offset),
+                int(chunk_length),
+                int(len(current_chunk))
+            ))
+            current_chunk = []
+    
+    with open(output_filename, 'rb') as fd:
+        write_output_file(output_filename, fd, output_base_url)
+    
+    os.unlink(output_filename)
+
+
+    final_files = write_partition_with_global_seq(partition_id, index_entries, num_lines, output_base_url)
+
+    return final_files
+
+def write_partition_with_global_seq(idx, partition_iter, records_per_partition=None, output_base_url=None):
+    partition_idx_file = f"idx-{idx:05d}.idx"
+    
+    # Calculate starting sequence number for this partition
+    start_seq = (idx * records_per_partition) + 1 if records_per_partition else 1
+    
+    with open(partition_idx_file, 'w') as f:
+        seq = start_seq
+        for record in partition_iter:
+            min_surt, _, filename, _, offset, length, _ = record
+            timestamp = "20240522010826" # TODO: what timestamp should this be????
+            f.write(f"{min_surt} {timestamp}\t{filename}\t{offset}\t{length}\t{seq}\n")
+            seq += 1
+    
+    with open(partition_idx_file, 'rb') as fd:
+        write_output_file(partition_idx_file, fd, output_base_url)
+    
+    os.unlink(partition_idx_file)
+
+    return [(partition_idx_file,True)]
+
+class ZipNumClusterCdx(CCFileProcessorSparkJob):
+    name = 'ZipNumClusterCdx'
+
+    def add_arguments(self, parser):
+        super(CCFileProcessorSparkJob,self).add_arguments(parser)
+        parser.add_argument("--output_base_url", required=False,
+                            default='my_cdx_bucket',
+                            help="destination for output")
+        parser.add_argument("--partition_boundries_file", required=False,
+                            help="Full path to a json file containing partition boundaries. if specified, and does not exist, will be created, otherwise, will be used.")
+        parser.add_argument("--num_lines", type=int, required=False,
+                            default=3000,
+                            help="number of lines to compress in each chunk")
+        parser.add_argument("--num_output_partitions", type=int, required=False,
+                            default=300,
+                            help="number of partitions/shards")
+
+    def run_job(self, session):
+        os.makedirs(self.args.output_base_url, exist_ok=True)
```

Comment:
See below: if `--output_base_url` is `file:/absolute/path/`, on Linux a relative folder `file:` is created in the current directory.

Reply:
I will double check that - it's not happening for me, but I may have omitted file:, so, may be testing it differently